server/internal/server/server.go

```diff
@@ -101,3 +126,144 @@ func (s *S) Run(ctx context.Context, port int, authConfig config.AuthConfig) err
 func (s *S) Stop() {
 	s.srv.Stop()
 }
+
+// RunRoute updates the routes periodically and upon pod events.
+func (s *S) RunRoute(ctx context.Context, errCh chan error) error {
+	k8sClient, err := k8s.NewClient()
```

Comment:
Can't we define a `Router` struct and move the logic there? We can also have a separate package (e.g., `route` or `router`).

Reply:
Can't we also have some unit test?

Reply:
Will add unit test later